libfwupd/generate-version-script.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+# pylint: disable=invalid-name,missing-docstring
+#
+# Copyright (C) 2017 Richard Hughes <richard@hughsie.com>
+#
+# Licensed under the GNU General Public License Version 2
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import xml.etree.ElementTree as ET
+
+def usage(return_code):
+    """ print usage and exit with the supplied return code """
+    if return_code == 0:
+        out = sys.stdout
+    else:
+        out = sys.stderr
+    out.write("usage: %s <NAME> <INPUT> <OUTPUT>\n" % sys.argv[0])
+    sys.exit(return_code)
+
+class LdVersionScript:
+    """ Rasterize some text """
+
+    def __init__(self, library_name):
+        self.library_name = library_name
+        self.releases = {}
+
+    def _add_node(self, node):
+        identifier = node.attrib['{http://www.gtk.org/introspection/c/1.0}identifier']
+        if 'version' not in node.attrib:
+            print('No version for', identifier)
+            sys.exit(1)
+            return
+        version = node.attrib['version']
+        if version not in self.releases:
+            self.releases[version] = []
+        release = self.releases[version]
+        release.append(identifier)
+        return version
+
+    def import_gir(self, filename):
+        tree = ET.parse(filename)
+        root = tree.getroot()
+        for ns in root.findall('{http://www.gtk.org/introspection/core/1.0}namespace'):
```

Comment:
Given you're reusing `"http://www.gtk.org/introspection/core/1.0"` over and over, maybe define that at the beginning of the function and use something like `"{%s}%" % (URL, "namespace")` to make this more readable and various lines less long

Reply:
Fixed.